APMrover2/sailboat.cpp

```diff
@@ -211,11 +224,26 @@ void Sailboat::get_throttle_and_mainsail_out(float desired_speed, float &throttl
         // linear interpolate mainsail value (0 to 100) from wind angle mainsail_angle
         float mainsail_base = linear_interpolate(0.0f, 100.0f, mainsail_angle,sail_angle_min,sail_angle_max);
 
-        // use PID controller to sheet out
-        const float pid_offset = rover.g2.attitude_control.get_sail_out_from_heel(radians(sail_heel_angle_max), rover.G_Dt) * 100.0f;
-
         mainsail_out = constrain_float((mainsail_base + pid_offset), 0.0f ,100.0f);
     }
+
+    //
+    // wingsail control
+    //
+
+    // wing sails auto trim, we only need to reduce power if we are tipping over
+    wingsail_out = 100.0f - pid_offset;
```

Comment:
Does this mean that the wing sail is nearly always fully left or right?  I'm probably misunderstanding something..

Reply:
yeah, it is for use with a sail such as [this ](https://www.wired.com/images_blogs/autopia/2014/02/ff_saildrone_f.jpg) full elevator gives full power either tack, there is no need to trim it for wind direction.

Reply:
it could be that for sailing upwind less elevator is more efficient, but this is probably to nuanced to implement until we do some proper testing

Reply:
The elevator is really there to aim the sail right?  It's not the wagging of the sail that provides the thrust of course?  If the elevator is for aiming the sail then I think the elevator output should reach an equilibrium and that would mostly not be full left or right always.

Reply:
exactly, the elevator just points the sail. The idea is that the user would set the servo min and max to a good equilibrium point for sailing on either tack.

Reply:
I also strongly suspect this will need to be changed after a bit of testing.  Surely it's going to stall or turn the sail too far so the wind hits the back of the wing or something like this.  Anyway, I won't hold up the PR for this 'cuz as you say, it's a start.